pr/ko/resolve-show-promise-invalid-details/pr.js

```diff
@@ -0,0 +1,107 @@
+/* global done:false */
+/* global error:false */
+/* global PaymentRequest:false */
+
+/**
+ * Initializes the payment request object.
+ */
+function buildPaymentRequest() {
+  if (!window.PaymentRequest) {
+    return null;
+  }
+
+  var supportedInstruments = [
+    {
+      supportedMethods: ['basic-card'],
+    },
+  ];
+
+  var details = {
+    total: {
+      label: 'Donation',
+      amount: {
+        currency: 'USD',
+        value: '1.00',
+        pending: true,
+      },
+    },
+  };
+
+  var request = null;
+
+  try {
+    request = new PaymentRequest(supportedInstruments, details);
+    if (request.canMakePayment) {
+      request
+        .canMakePayment()
+        .then(function(result) {
+          info(result ? 'Can make payment' : 'Cannot make payment');
+        })
+        .catch(function(err) {
+          error(err);
+        });
+    }
+  } catch (e) {
+    error("Developer mistake: '" + e + "'");
+  }
+
+  return request;
+}
+
+var request = buildPaymentRequest();
+
+/**
+ * Launches payment request for credit cards.
+ */
+function onBuyClicked() {
+  // eslint-disable-line no-unused-vars
+  if (!window.PaymentRequest || !request) {
+    error('PaymentRequest API is not supported.');
+    return;
+  }
+
+  try {
+    request
+      .show(
+        new Promise(function(resolve) {
+          info('Calculating final price...');
+          window.setTimeout(function() {
+            info('The final price is $2.00 USD.');
+            resolve({
+              total: {
+                label: 'Donation',
+                amount: {
+                  currency: 'USD',
+                  value: '-2.00',
```

Comment:
Add an inline comment:
// Negative total is invalid.

Reply:
Done.